### PR TITLE
LUCENE-9488 Update release process to work with gradle

### PR DESCRIPTION
**Pull request by madrob**
https://issues.apache.org/jira/browse/LUCENE-9488

### CODE REVIEW

File: dev-tools/scripts/releaseWizard.yaml [hunk 1]
```diff
@@ -411,7 +410,7 @@ groups:
         cmd: git pull
         stdout: true
       - !Command
-        cmd: ant clean precommit
+        cmd: ./gradlew clean check -x test
```
Comment (janhoy) [inline dev-tools/scripts/releaseWizard.yaml:413]: Will the `./` syntax work on Windows?

**Review by janhoy — COMMENTED** [review]
Did not test the changes, have only one comment regarding Windows compatibility

File: dev-tools/scripts/releaseWizard.yaml (hunk 1 above)
Comment (madrob) [reply dev-tools/scripts/releaseWizard.yaml:413]: It works in PowerShell but not Command Prompt. I don't know how important this distinction is.

**Comment by HoustonPutman** [comment]
Should the smokeTester be in the scope of this PR?

**Comment by madrob** [comment]
@janhoy I'm trying to run this, but the `create_stable_branch` step tries to create a new `branch_8x` for me, I think that logic needs some updating, but I'm not completely sure what needs to change. If you're able, please take a look, otherwise I'll skip that part and come back to it later.

File: dev-tools/scripts/releaseWizard.yaml (hunk 1 above)
Comment (janhoy) [reply dev-tools/scripts/releaseWizard.yaml:415]: You can definere a variable like we do for rename command, that used `gradlew.bat` for win?

**Comment by janhoy** [comment]
There may be bugs related to major release since that has never been done with the wizard. I guess it should have created a `branch_9x`?

File: dev-tools/scripts/releaseWizard.yaml (hunk 1 above)
Comment (madrob) [reply dev-tools/scripts/releaseWizard.yaml]: Works great!

**Comment by madrob** [comment]
@janhoy This isn't actually ready yet, but I'd like to merge what I have so far. The reason is that the way the script pulls from upstream to prep the RC, I kind of need changes on the actual master branch in order to proceed with further testing without resorting to some super hacky tricks locally to make it think that my fork is the place to pull from.

**Review by janhoy — APPROVED** [review]

**Comment by janhoy** [comment]
Sure, all progress is good progress

File: lucene/version.properties [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# This file contains some version properties as used by various build files.
```
Comment (uschindler) [inline lucene/version.properties:1]: Ülease remove that file, it wont be used.

If release scripts need to edit the version number after a release, <root>/build.gradle is your friend. Theres also a version.base and that's the source of information.

https://github.com/apache/lucene-solr/blob/master/build.gradle#L38

There is a reason why this file was removed, so dont restore it!

**Review by uschindler — COMMENTED** [review]
Please revert the file addition.

File: dev-tools/scripts/buildAndPushRelease.py [hunk 3]
```diff
@@ -108,37 +107,26 @@ def prepare(root, version, gpgKeyID, gpgPassword):
...
+  print('  ./gradlew -Dtests.badapples=false clean check')
+  run('./gradlew -Dtests.badapples=false clean check')
 
   open('rev.txt', mode='wb').write(rev.encode('UTF-8'))
   
-  print('  lucene prepare-release')
-  os.chdir('lucene')
-  cmd = 'ant -Dversion=%s' % version
+  print('  prepare-release')
+  cmd = './gradlew -Dversion=%s clean' % version
```
Comment (uschindler) [inline dev-tools/scripts/buildAndPushRelease.py:116]: This must be `-Dversion.release=%s`, as gradle has some reserved meaning of `version`.

File: dev-tools/scripts/releaseWizard.yaml [hunk 4]
```diff
@@ -1059,7 +1056,7 @@ groups:
         logfile: checkout-release-tag.log
         tee: true
       - !Command
-        cmd: ant documentation -Dversion={{ release_version }}
+        cmd: "{{ gradle_cmd }} documentation -Dversion={{ release_version }}"
```
Comment (uschindler) [inline dev-tools/scripts/releaseWizard.yaml:1059]: This must be `-Dversion.release=%s`, as gradle has some reserved meaning of `version`.

**Review by uschindler — COMMENTED** [review]
Please change the property names, too.

**Pull request merged**